src/sagemaker/workflow/emr_step.py

```diff
@@ -198,9 +208,16 @@ def __init__(
         if cluster_id is not None and cluster_config is not None:
             raise ValueError(ERR_STR_WITH_BOTH_CLUSTER_ID_AND_CLUSTER_CFG.format(step_name=name))
 
+        if execution_role_arn is not None and cluster_config is not None:
+            raise ValueError(ERR_STR_WITH_BOTH_EXEC_ROLE_ARN_AND_CLUSTER_CFG.format(step_name=name))
```

Comment:
Is this something we need to validate? This seems like a requirement from EMR side right? Should we start the EMR job and let it do this validation instead?

Reply:
We do this validation in the DSL. I used the same pattern that we are enforcing for `cluster_id` and `cluster_config`: for these, we are checking both in the SDK and in the DSL that they are not both set.

Reply:
Out of curious, do you know is there any reason we need to do this checking in both places? 
If the DSL validation already covers these and they happen during creating pipeline time, we don't need to repeat the same validation in SDK. 
However, I'm fine to follow the same pattern that we already have for cluster_id and cluster_config in this PR.